### DISCUSSION TITLE
Accessing SystemVerilog interface modport array elements

### DISCUSSION THREAD

**Discussion by matthuszagh** [cocotb/cocotb#3430] [Q&A]
I have a module of the form

```
module mod #(
    parameter integer N = 8,
    // ...
) (
    iface.port1 a[N],
    // ...
);

    // ...

endmodule
```

where `iface` is a SystemVerilog interface providing the modport `port1`.

In cococtb, I'd like to be able to do something like `dut.a[0]`. But, when I attempt that I get the error: `TypeError: 'HierarchyObject' object is not subscriptable`. Is there any way to accomplish what I'm trying here?

I'm using questa sim as the underlying simulator. Thanks!

**Comment by ktbarrett** [comment]
This seems like something not working as intended. I suggest you open an issue and fill out everything the template asks, then we can go about debugging the issue.
